### PR TITLE
feature: inline alert component

### PR DESCRIPTION
**Pull request by steven-klein**
# What this does

- adds the `InlineAlert` component

## Notes

- styling is mostly a copy/paste from the existing alert in Archive/Tailwind UI
- prop names diverge a bit from what's currently in use in Portal and Archive and does not support slots
- component API attempts to stay close to CTACard for consistency, but does diverge a bit

### General Types

Screenshot 2023-10-13 at 4 49 16 PM

### Variations via Props

Screenshot 2023-10-16 at 10 23 25 AM

### CODE REVIEW

File: src/lib-components/indicators/InlineAlert.vue [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  CheckCircleIcon,
+  InformationCircleIcon,
+  ExclamationIcon,
+  XCircleIcon,
+} from "@heroicons/vue/solid"
+
+import { XIcon } from "@heroicons/vue/outline"
+
+const props = withDefaults(
+  defineProps<{
```
Comment (steven-klein) [inline src/lib-components/indicators/InlineAlert.vue:14]: There's a bit of deviation from existing patterns here on the props.  

- I preferred `type` over `kind`, but am ultimately indifferent if there are strong opinions in either direction.
- Shortening `primaryBtnLink` to `btnLink`, etc. was preferred as I'm mostly anticipating a single CTA on alerts when used.

File: src/lib-components/indicators/InlineAlert.vue [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    btnAction: "",
+    btnClose: "",
+    contentColor: "",
+    iconColor: "",
+    titleColor: "",
+  }
+
+  switch (props.type) {
+    case "alert":
+      display.bgColor = "bg-red-50"
```
[the remark this answers is not in the document]
Comment (profsmallpine) [reply src/lib-components/indicators/InlineAlert.vue:58]: Yes! 🤦

**Review by profsmallpine — COMMENTED** [review]
Looks good, 2 minor nits

File: src/lib-components/indicators/InlineAlert.vue [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+      display.iconColor = "text-red-400"
+      display.titleColor = "text-red-800"
+      break
+    case "info":
+      display.bgColor = "bg-blue-50"
+      display.btnAction =
+        "bg-blue-50 text-blue-900 hover:bg-blue-100 focus:ring-blue-600 focus:ring-offset-blue-50"
+      display.btnClose =
+        "bg-blue-50 text-blue-600 hover:bg-blue-100 focus:ring-blue-600 focus:ring-offset-blue-50"
+      display.contentColor = "text-blue-800"
```
[the remark this answers is not in the document]
Comment (steven-klein) [reply src/lib-components/indicators/InlineAlert.vue:71]: Nope, I started out with slightly different color weights on them, but no reason to dupe now.

**Comment by steven-klein** [comment]
> Is the intention this replaces `<InlineFlash />` in Portal?

Yes, we're running two different versions of effectively the same component between portal and archive.  This aims to align them.

**Pull request merged**